src/tests/syncTests.ts

```diff
@@ -87,6 +87,12 @@ export class SyncTests {
     const coveredApexClassIdSet = new Set<string>();
     const { apexTestClassIdSet, testResults } =
       this.buildSyncTestResults(apiTestResult);
+    const regularTests = testResults.filter(
```

Comment:
@CristiCanizales because of the original poor code structure, I can see why you would adapt this function to produce the correct TestResult structure. However, this introduces more duplicate code and defeats the purpose of the transform you added to async test.

Please delegate transform in this method as well.

I would also like you to think about reconciling some of the duplicate code found in this method and formatAsyncResult.